src/auth/src/credentials/idtoken.rs

```diff
@@ -12,10 +12,49 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//! Obtain [OIDC ID Tokens].
+//!
+//! `IDTokenCredentials` provide a way to obtain OIDC ID tokens, which are
+//! commonly used for [service to service authentication], like when services are
+//! hosted in Cloud Run or mediated by Identity-Aware Proxy (IAP).
+//! Unlike access tokens, ID tokens are not used to authorize access to
+//! Google Cloud APIs but to verify the identity of a principal.
+//!
+//! This module provides `IDTokenCredentials` which serves as a wrapper around
+//! different credential types that can produce ID tokens, such as service
+//! accounts or metadata server credentials.
```

Comment:
Didn't you say that above?  maybe:


```suggestion
//! The main type in this module is [IDTokenCredentials].  This is an opaque type
//! that implements the ... trait and can be used to obtain OIDC ID tokens.  Use the
//! builders in each submodule to create `IDTokenCredentials` based on different
//! token sources. 
```

Or maybe "authentication methods" instead of "token sources"?